### PR TITLE
Walk/run animation blending

### PR DESCRIPTION
**Pull request by skaljac**
Issue related to this PR:
https://github.com/webaverse/app/issues/1695

Steps for testing:
Walk around
Press Shift to start running
Switch from walk to run and vice versa should not be instant

### CODE REVIEW

**Comment by skaljac** [comment]
https://user-images.githubusercontent.com/4487857/145001320-839324d5-03ae-4ab4-bb67-2aaae43d140b.mp4

**Review by yaserOSource — APPROVED** [review]
Code looks good to me. 

Will test

**Comment by amnaarshed4224** [comment]
Device:
Dell precision 7710
Ram: 16gb
Ci7 6th gen
Nvidia Quadro m5000m (60Hz refresh rate)

Issues:

1-Walk to run is smooth now but run to walk appears quite instantaneous and a bit jerky.

https://user-images.githubusercontent.com/35926530/145093653-ab03f4d3-1f9d-42db-926e-b94cf974a5d2.mp4

2-Walk to run is smooth as whole but while closely looking at footsteps while the blend happens, The footsteps appear a bit off and different than original flow. They appear to be out of synchronization for a moment.

https://user-images.githubusercontent.com/35926530/145093702-19c03fa1-15e9-4711-afdc-1219ba13e4ad.mp4


3-While we start with run(By pressing shift+W) instead of first walking and then shifting to run, There appears an instant jerk at the start.

https://user-images.githubusercontent.com/35926530/145093771-b0f2387b-3ac4-4e2d-8136-5e3a835907d8.mp4

4-At low fps of around 25-40, The avatar slows down while running. Sometimes it doesn't run at low fps.

https://user-images.githubusercontent.com/35926530/145093812-5e65abf7-1728-4967-8292-9cc1911c7ca7.mp4


https://user-images.githubusercontent.com/35926530/145093798-016a7113-c768-494b-ae22-f6721bdea904.mp4

File: avatars/avatars.js
Comment (avaer) [inline avatars/avatars.js:1647]: This looks dependent on the frame time.

Comment (skaljac) [reply avatars/avatars.js:1647]: I should have use fimeDiff in this calculation

File: avatars/avatars.js
Comment (avaer) [inline avatars/avatars.js:1723]: This is not the right way to do it. We can smoothly blend on a curve over the 
instantaneous velocity, not an if statement plus a lerp to switch between walk/run modally.

**Comment by skaljac** [comment]
I have resolve main issue with this code. It is not modal. It is smooth.

Works as expected.

Also I had to introduce float speedValue (it is lerping to speed value, it respects timeDiff) beacuse there is no gradual change in speed between walk and run.

**Comment by skaljac** [comment]
https://user-images.githubusercontent.com/4487857/145305180-82de2981-b67a-4dac-8e76-d072c92e5f49.mp4

**Review by avaer — APPROVED** [review]

**Comment by amnaarshed4224** [comment]
1-Crouch run needs improvement as The transition from standing to walk and vice versa is jerky.


https://user-images.githubusercontent.com/35926530/145470688-5a5918cf-0eef-435b-9517-8d3ce2a3eabe.mp4


2-The footsteps at the start of walk/run transition seem a bit off and needs improvement. The pr overall has improved the walk/run transition and can be merged after crouch walk is fixed. This bug can be worked on in future.


https://user-images.githubusercontent.com/35926530/145470708-e7bd140d-2550-41db-b16f-3901d3e6d87a.mp4

**Comment by avaer** [comment]
This needs a rebase after merging https://github.com/webaverse/app/pull/1856

**Comment by skaljac** [comment]
Rebase is done

**Comment by avaer** [comment]
Made some major timing fixes, blending should be smooth now.

**Pull request merged**
